### DISCUSSION TITLE
Does Peptidestubmover renumber residues in the pose?

### DISCUSSION THREAD

**Discussion by tsamdin** [RosettaCommons/rosetta#130] [Rosetta] [answered]
I'm using the Generalized KIC Demos as a guide for redesigning a loop segment in a protein of interest. 

In the first tutorial, the input structure has continuous numbering despite having residues that would take the place of the loop removed. As peptidestubmover inserts and appends residues, does the numbering of the chains downstream of the new loop residues get updated as well? 

On a related note does the input structure have to have continuous numbering? Could I submit a single chain structure that is numbered 1-10 and 15-20 where residues 11-14 are missing, but will be added by peptidestubmover and closed with Gen KIC?

**Comment by vmullig** [comment, marked as answer]
Sorry for the slow reply.  Yes, the PeptideStubMover inevitably renumbers the pose.  Rosetta has two pose numbering systems: the internal pose numbering is hard-coded to start at 1 and to number residues consecutively and uniquely.  In addition, a pose _can_ (but does not necessarily) store PDB numbers read in from a PDB file, but what happens when residues are inserted or appended can be ambiguous.  Most protocols operate on Rosetta numbers.  The PeptideStubMover, when appending residues, will continue the numbering (so if your pose consists of residues 1 through 100 and you add five more residues, these will be residues 101, 102, 103, 104, and 104).  When prepending or inserting residues, downstream residue are renumbered (so if you have a pose consisting of residues 1-50 in chain A and 51-100 in chain B, and you prepend 5 residues to chain B, the new residues are residues 51-55, and the residues that were formerly residues 51-100 are now residues 56-105).

**Comment by vmullig** [comment]
> On a related note does the input structure have to have continuous numbering? Could I submit a single chain structure that is numbered 1-10 and 15-20 where residues 11-14 are missing, but will be added by peptidestubmover and closed with Gen KIC?

With Rosetta numbering, yes, the numbering must be contiguous.  You _could_ have a pose which, on input, is residues 1-15, and which subsequently has 5 residues inserted by the PeptideStubMover, making it [1-10](11-15)[16-20] (where square brackets represent the old sequence and round brackets represent the insertion from the PeptideStubMover), and you could subsequently close this with GenKIC.  But you have to be careful to write your protocol so that at any given time, you're referring to the _current_ numbering -- i.e. treating the second segment as 10-15 before the application of the PeptideStubMover, and 16-20 after applying it.

**Reply by tsamdin** [reply]
That clears up a lot! Thank you
